src/controllers/cart.controller.js

```diff
@@ -0,0 +1,96 @@
+const isValidObjectId = require("../utils/isObjectId.util");
+const Carts = require("../models/cart.model");
+const asyncHandler = require("../utils/asyncHandler.util");
+
+const getCart = asyncHandler(async (req, res) => {
+    const cart = await Carts.findOne({ user: req.user._id }).populate('products.product');
+    res.json(cart?.products || []);
```

Comment:
I think || [] is required because if no product is present it will itself return an empty array anyway :) 
Check this out :)

Reply:
But if the cart is not present then the above cart would return null.. and products of null would return undefined.. and then i'd have to put a check in the frontend. 😅